### PR TITLE
[MC,ELF] Use loc from the directive for `.abort`

### PR DESCRIPTION
**Pull request by chestnykh**

### CODE REVIEW

File: llvm/lib/MC/MCParser/AsmParser.cpp [hunk 1]
```diff
@@ -5095,18 +5095,16 @@ bool AsmParser::parseDirectiveComm(bool IsLocal) {
...
   StringRef Str = parseStringToEndOfStatement();
   if (parseEOL())
     return true;
 
   if (Str.empty())
-    return Error(Loc, ".abort detected. Assembly stopping.");
+    return Error(DirectiveLoc, ".abort detected. Assembly stopping.");
   else
-    return Error(Loc, ".abort '" + Str + "' detected. Assembly stopping.");
+    return Error(DirectiveLoc,
```
Comment (s-barannikov) [inline llvm/lib/MC/MCParser/AsmParser.cpp:5106]: While here, can you remove the `else`? https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return

The `return` below is unreachable, please remove it.

Comment (chestnykh) [reply llvm/lib/MC/MCParser/AsmParser.cpp]: Done, thx for noticing this

File: llvm/lib/MC/MCParser/AsmParser.cpp [hunk 2]
```diff
@@ -5095,21 +5095,16 @@ bool AsmParser::parseDirectiveComm(bool IsLocal) {
...
+bool AsmParser::parseDirectiveAbort(SMLoc DirectiveLoc) {
   StringRef Str = parseStringToEndOfStatement();
   if (parseEOL())
     return true;
 
   if (Str.empty())
-    return Error(Loc, ".abort detected. Assembly stopping.");
-  else
-    return Error(Loc, ".abort '" + Str + "' detected. Assembly stopping.");
-  // FIXME: Actually abort assembly here.
```
Comment (s-barannikov) [inline llvm/lib/MC/MCParser/AsmParser.cpp:5110]: This FIXME hasn't actually been fixed

Comment (chestnykh) [reply llvm/lib/MC/MCParser/AsmParser.cpp:5110]: Sorry, accidentally removal :)

**Comment by chestnykh** [comment]
pipeline error seems to be unrelated:
`fatal error: 'llvm/ADT/DenseMapInfo.h' file not found`

File: llvm/lib/MC/MCParser/AsmParser.cpp (hunk 2 above)
Comment (MaskRay) [reply llvm/lib/MC/MCParser/AsmParser.cpp:5110]: While here, please drop the trailing period to make the diagnostics more conventional

Comment (chestnykh) [reply llvm/lib/MC/MCParser/AsmParser.cpp:5110]: Done

Comment (s-barannikov) [reply llvm/lib/MC/MCParser/AsmParser.cpp:5110]: The last commit removed the FIXME again

Comment (chestnykh) [reply llvm/lib/MC/MCParser/AsmParser.cpp:5110]: Done

**Review by MaskRay — APPROVED** [review]

**Pull request merged**